cpp/src/io/parquet/page_data.cu

```diff
@@ -971,6 +996,11 @@ static __device__ bool setupLocalPageInfo(page_state_s* const s,
         s->dtype_len = s->dtype_len <= sizeof(int32_t)   ? sizeof(int32_t)
                        : s->dtype_len <= sizeof(int64_t) ? sizeof(int64_t)
                                                          : sizeof(__int128_t);
+      } else if (s->col.converted_type == DECIMAL && data_type == BYTE_ARRAY) {
+        auto const decimal_precision = s->col.decimal_precision;
+        s->dtype_len = decimal_precision <= MAX_DECIMAL32_PRECISION   ? sizeof(int32_t)
+                       : decimal_precision <= MAX_DECIMAL64_PRECISION ? sizeof(int64_t)
+                                                                      : sizeof(__int128_t);
```

Comment:
could this be done in the switch above? The only different I see is in the value of `s->dtype_len_in`, which is probably irrelevant for byte arrays.